ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/QueryStreamWriter.java

```diff
@@ -124,6 +125,21 @@ private void write(OutputStream output, GenericRow row) throws IOException {
     output.flush();
   }
 
+  private void drainAndThrowOnError(final OutputStream out) throws Throwable {
+    if (streamsException == null) {
+      return;
+    }
+
+    final List<KeyValue<String, GenericRow>> rows = Lists.newArrayList();
+    queryMetadata.getRowQueue().drainTo(rows);
+
+    for (final KeyValue<String, GenericRow> row : rows) {
+      write(out, row.value);
```

Comment:
We might get a null GenericRow here. We should check for that and skip those rows in the output.

Reply:
Ah I see now. I was worried about the case where a query legitimately produces null output (e.g. a HAVING clause evaluates to false, or a basic projection has a null record in the source). But looks like those are filtered out by QueuedSchemaKStream. So we're all good here!